code/processes/tailsort.q

```diff
@@ -0,0 +1,77 @@
+\d .ts
+
+taildir:hsym `$getenv`KDBTAIL;                                             /-load in taildir env variables
+hdbdir:hsym `$getenv`KDBHDB;                                               /-load in hdb env variables
+savelist:@[value;`savelist;`quote`trade];                                  /-list of tables to save to HDB
+taildbs:key taildir;                                                       /-list of tailDBs that need saved to HDB
+taildirs:();                                                               /-empty list to append tailDB paths to - to be used
+                                                                           / when HDB save is complete to delete tailDB partitions
+\d .
+
+savescompleted:0;                                                          /-variable to count how many tailDBs have been saved to HDB
+
+endofday:{[pt;procname]
+  /- function to trigger data load & save to HDB once endofday message is received from tailer(s)
+  .lg.o[`endofday;"end of day message received from ",string[procname]," - ",string[pt]];
+  loadandsave[pt;procname];
+  };
+
+loadandsave:{[pt;procname]
+  /-function to merge tables from subpartitions in tailDB and save to HDB
+  taildir:` sv (.ts.taildir;procname;`$string pt);
+  .ts.taildirs,:taildir;
+  /-merge tables from tailDBs and save to HDB
+  mergebypart[taildir;pt;;.ts.hdbdir] each .ts.savelist;
+  /-increase savescompleted counter
+  savescompleted+::1;
+  .lg.o[`sortcomplete;"end of day sort complete for ",string[procname]];
+  /-check if all eod saves have been completed, if so trigger savecomplete
+  if[savescompleted = count .ts.taildbs;savecomplete[pt;.ts.savelist]];
```

Comment:
if I'm reading this right, this means we can only have one tailsort process which will do EOD sort for all taildbs? assume that is intended and having multiple tailsort procs to parallelise EOD sort will be a future enhancement?

Reply:
yeah with this setup we will only be able to have one tailsort process to do all the work and adding secondary processes to parallelise the sort is a new jira ticket that will need done - will probably need to rethink the counter method for secondary processes but I'll detail this in the ticket

Reply:
Sure, makes sense. Thanks